### DISCUSSION TITLE
Assert seems to not work during configure

### DISCUSSION THREAD

**Discussion by SMorettini** [nasa/fprime#1919] [Q&A] [answered]
In my code, I had the following assert failing:
```
Assert file "Fprime/Svc/ActiveRateGroup/ActiveRateGroup.cpp": Line: 35 10 12
```
It is caused because I changed the number of ports for the rate group but I didn't increase the context. But that's not the point of my question. 

The assert is verified by the executable continues to run. After a while I made another assert failing and in that case the code stop running with a segmentation fault as expected.

Is it by design that the asserts in the configure will not make the executable stop or it is a bug?

**Comment by timcanham** [comment]
The design for the assert logic is that a project will decide how to respond the asserts in a way that makes sense for them. Some projects wish to do a seg fault, some wish to suspend the offending task so it can be examined, and some wish to do a hardware reset. For that reason, we designed assert to follow an assert hook registration pattern:

https://nasa.github.io/fprime/UsersGuide/dev/assert.html

A common pattern is to use `Svc/AssertFatalAdapter` as the assert hook. That turns it in to a FATAL event that gets processed by `Svc/ActiveLogger`. That will give it a chance to be reported outside the processor (like a radio) prior to recovery so operators know what happened. If it just goes to the console and the evidence is lost, it can be hard to debug. `Svc/ActiveLogger` has a special FATAL port (`FatalAnnounce`) that gets called on the same thread that reports the FATAL, in this case the assert. The project can write a special component to connect to the `FatalAnnounce` port to do a custom response. Some projects we've done pause for a second or two to allow time for the FATAL to be emitted, and then do a hardware reset.

The `Ref` example connects `Svc/FatalHandler` to this port, which in a Linux deployment should raise `SIGSEV` and then exit.

**Comment by bocchino** [comment, marked as answer]
You may be running into an issue that was fixed in this PR: https://github.com/nasa/fprime/pull/1865. Before this fix, assertion failures would not cause a halt if they occurred before the assertion ports were hooked up.

**Reply by SMorettini** [reply]
I tried that code in my software and now the assert work as expected. Thanks for pointing it out, I didn't notice the PR.
